spylib/oauth/types.py

```diff
@@ -0,0 +1,111 @@
+from typing import List
+
+from pydantic import BaseModel, validator
+
+
+def _parse_scope(scope: str) -> List[str]:
+    return scope.split(',')
+
+
+class OfflineTokenModel(BaseModel):
+    """
+    [Read more about Offline access](https://shopify.dev/apps/auth/oauth/access-modes#offline-access)
+    """
+
+    access_token: str
+    """
+    An API access token that can be used to access the shop's data as long as your app is installed. Your app should store the token somewhere to make authenticated requests for a shop’s data.
+    """
+
+    scope: List[str]
+    """
+    The list of access scopes that were granted to your app and are associated with the access token.
+    """
+
+    _normalize_scope = validator('scope', allow_reuse=True, pre=True)(_parse_scope)
+
+
+class AssociatedUser(BaseModel):
+    """
+    Shopify staff user associated with an online token.
+    """
+
+    id: int
+    """
+    The user id.
+
+    [Can be associated with the `sub` of a session token](https://shopify.dev/apps/auth/oauth/session-tokens#payload)
+    """
+
+    first_name: str
+    """
+    The user's first name.
+    """
+
+    last_name: str
+    """
+    The user's last name.
+    """
+
+    email: str
+    """
+    The user's email.
+    """
+
+    email_verified: bool = False
+    """
+    If the user's email has been verified.
+
+    If `False` or `None` email cannot be trusted (legacy accounts).
+    """
```

Comment:
Comment says that `None` is a possibility but the type doesn't match:
```suggestion
    email_verified: Optional[bool] = False
    """
    If the user's email has been verified.

    If `False` or `None` email cannot be trusted (legacy accounts).
    """
```

Reply:
Hmm, so probably a documentation issue. It can be none in very niche cases on very old accounts but I'm not sure we want to represent that in the model?

I was hoping with the default false the optional could be avoided